tests/Installer/PackagesInstallerTest.php

```diff
@@ -11,121 +11,131 @@
-                ->expects(self::exactly(count($expected)))
-                ->method('installPackage')
-                ->withConsecutive(...$expected);
-        } else {
-            $depInstaller
-                ->expects(self::never())
-                ->method('installPackage');
-        }
+        $depInstaller
+            ->expects($this->atLeastOnce())
+            ->method('installPackage');
 
         $installer->install();
     }
 
     /**
-     * @return array
+     * @throws Exception
      */
-    public function dataProvider(): array
+    public function testCanMergeDefaultPackagesWhenInstalling(): void
     {
-        return [
-            [
-                'magento1',
-                $this->createLinkMocks(['foo/bar']),
-                [['youwe/coding-standard-magento1']]
-            ],
-            [
-                'magento1',
-                $this->createLinkMocks(
-                    ['foo/bar', 'youwe/coding-standard-magento1']
-                ),
-                [['youwe/coding-standard-magento1']]
-            ],
-            [
-                'magento2',
-                $this->createLinkMocks(['foo/bar']),
-                [['youwe/coding-standard-magento2']]
+        $composer     = $this->createMock(Composer::class);
+        $typeResolver = $this->createMock(ProjectTypeResolver::class);
+        $depInstaller = $this->createMock(DependencyInstaller::class);
+        $io           = $this->createMock(IOInterface::class);
+
+        // Simulate magento 1 project
+        $typeResolver
+            ->method('resolve')
+            ->willReturn('magento1');
+
+        $mapping = [
+            MappingResolver::DEFAULT_MAPPING_TYPE => [
+                [
+                    'name' => 'phpunit/phpunit',
+                    'version' => '@stable',
+                    'dev' => true,
+                    'allowVersionOverride' => false
+                ]
             ],
-            [
-                'default',
-                $this->createLinkMocks(['foo/bar']),
-                null
+            'magento1' => [
+                [
+                    'name' => 'youwe/coding-standard-magento1',
+                    'version' => '^1.3.0',
+                    'dev' => true,
+                    'allowVersionOverride' => true
+                ]
             ],
-            [
-                'unknown',
-                $this->createLinkMocks(['foo/bar']),
-                null
-            ]
         ];
+
+        $installer = new PackagesInstaller(
+            $composer,
+            $typeResolver,
+            $io,
+            $depInstaller,
+            $mapping
+        );
+
+        // We expect 2 packages to be installed due to the mapping merge
+        $depInstaller
+            ->expects($this->exactly(2))
+            ->method('installPackage');
+
+        $io
+            ->expects($this->exactly(2))
+            ->method('write')
+            ->willReturnOnConsecutiveCalls(
```

Comment:
FIXME: Mixup with consecutive argument checking, you're mocking a return here but `$io->write` will return nothing because it's declared as `@return void`